### PR TITLE
add the new BitChoiceGroup component (#1401)

### LINKED ISSUE #1401: Add new `BitChoiceGroup` component
The current `BitChoiceGroup` component is not implemented exactly based on the design of the corresponding component in the Fluent UI library. So it is decided to implement a new component with this name based on the mentioned component's API/UI design.

### PR DESCRIPTION
**Pull request by sajadkardel**
this closes #1401

### CODE REVIEW

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Components/ChoiceGroup/BitChoiceGroupDemo.razor.cs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+                DefaultValue = "null",
+                Description = "Callback for when the option has been changed"
+            },
+            new ComponentParameter
+            {
+                Name = "OnClick",
+                Type = "EventCallback<ChangeEventArgs>",
+                DefaultValue = "null",
+                Description = "Callback for when the option clicked"
+            },
```
Comment (Cyrus-Sushiant) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Components/ChoiceGroup/BitChoiceGroupDemo.razor.cs:161]: Add `Value` and `ValueChanged` to the list of parameter

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitChoiceGroupTests.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                });
+
+            var bitChoiceGroupOptions = component.Find(".bit-chgo");
+
+            var visualClass = visual == Visual.Cupertino ? "cupertino" : visual == Visual.Material ? "material" : "fluent";
+
+            if (groupIsEnabled is false || optionIsEnabled is false)
+            {
+                Assert.IsTrue(bitChoiceGroupOptions.ClassList.Contains($"bit-chgo-disabled-{visualClass}"));
+            }
```
Comment (Cyrus-Sushiant) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitChoiceGroupTests.cs:51]: I think it is better to check that it doesn't have class `disabled` when the above condition is false.
```csharp
else 
{
  Assert.IsFalse(bitChoiceGroupOptions.ClassList.Contains($"bit-chgo-disabled-{visualClass}"));
}
```

**Review by Cyrus-Sushiant — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ChoiceGroup/BitChoiceGroup.razor.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+        /// <summary>
+        /// Name of ChoiceGroup, this name is used to group each option into the same logical ChoiceGroup.
+        /// </summary>
+        [Parameter] public string Name { get; set; } = Guid.NewGuid().ToString();
+
+        /// <summary>
+        /// Default selected key for ChoiceGroup.
+        /// </summary>
+        [Parameter] public string? DefaultSelectedKey { get; set; }
```
Comment (Cyrus-Sushiant) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/ChoiceGroup/BitChoiceGroup.razor.cs:44]: Plz, change `DefaultSelectedKey` to `DefaultValue`

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ChoiceGroup/BitChoiceGroupOption.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bit.Client.Web.BlazorUI.Components.ChoiceGroup
+{
+    public class BitChoiceGroupOption
```
Comment (Cyrus-Sushiant) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/ChoiceGroup/BitChoiceGroupOption.cs:10]: Plz add `AriaLabel`, `LabelId` and `Id` prop like [fluent](https://developer.microsoft.com/en-us/fluentui#/controls/web/choicegroup)

**Review by Cyrus-Sushiant — CHANGES_REQUESTED** [review]

**Comment by msynk** [comment]
@Cyrus-Sushiant plz re-review

**Review by Cyrus-Sushiant — APPROVED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**